payjoin/src/receive/mod.rs

```diff
@@ -791,7 +803,32 @@ impl PayjoinProposal {
             self.payjoin_psbt.inputs[i].final_script_sig = None;
             self.payjoin_psbt.inputs[i].final_script_witness = None;
         }
-        Ok(self.payjoin_psbt)
+        Ok(PayjoinProposal {
+            payjoin_psbt: self.payjoin_psbt,
+            owned_vouts: self.owned_vouts,
+            params: self.params,
+        })
+    }
+
+    pub fn finalize_proposal(
+        mut self,
+        wallet_process_psbt: impl Fn(&Psbt) -> Result<std::string::String, Error>,
+        min_feerate_sat_per_vb: Option<u64>,
+    ) -> Result<PayjoinProposal, RequestError> {
+        let payjoin_proposal_psbt: &Psbt = self.apply_fee(min_feerate_sat_per_vb)?;
+        let payjoin_proposal_psbt = match wallet_process_psbt(payjoin_proposal_psbt) {
+            Ok(p) => p,
+            Err(_) => {
+                return Err(RequestError::from(InternalRequestError::PsbtFinalizingError(
+                    "wallet_process_psbt failed",
+                )));
+            }
+        };
```

Comment:
```suggestion
        let psbt = wallet_process_psbt(payjoin_proposal_psbt)
            .map_err(|_| InternalRequestError::PsbtFinalizingError.into())?;
```

I think it's fine to call it plain `psbt` in this context, we know it's a payjoin_proposal method function.